test/integration/PluginTests.ts

```diff
@@ -0,0 +1,69 @@
+import { TASK_FLATTEN_GET_FLATTENED_SOURCE } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { BuidlerRuntimeEnvironment } from "@nomiclabs/buidler/types";
+import { assert } from "chai";
+import { ethers } from "ethers";
+// @ts-ignore
+import * as linker from "solc/linker";
+
+import ContractCompiler from "../../src/ContractCompiler";
+import deployer from "../util/ContractDeployer";
+
+declare module "mocha" {
+  interface Context {
+    env: BuidlerRuntimeEnvironment;
+  }
+}
+
+describe("Plugin integration tests", function() {
+  this.timeout(100000);
+
+  let env: BuidlerRuntimeEnvironment;
+
+  before("Buidler project setup", function() {
+    process.chdir(__dirname + "/../buidler-project");
+    process.env.BUIDLER_NETWORK = "develop";
+    env = require("@nomiclabs/buidler");
+  });
+
+  it("Test verifying deployed contract on etherscan", async () => {
```

Comment:
I noticed this tests fail because no valid api key is defined. It would be nice to test an actual contract verification (on ropsten), and include it into the CI.

Reply:
Also, what happen if we reach the 100 daily submissions limit?

Reply:
I've changed CI to run only unit tests. These tests are so unstable. Etherscan sometimes needs more than 6-7 confirmations to reach block with deployed contract and this fails with error Contract source not found. Also there is constant need for watching if wallet has funds, and ci will be running for 4-5 mins, which is not ideal. Might be good solution to run it only on master or something.

There is no need for valid apikey on etherscan for testnets. Wallet private key is needed for this and can be suplied via travis env variable

Reply:
You are right @mpetrunic, running all the tests in the CI is more complicated than we initially thought.